### PR TITLE
feat(agg): support global topn

### PR DESCRIPTION
**Pull request by Shmiwy**
Signed-off-by: Shmiwy <wyf000219@126.com>

I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

support global stateless top n

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

**Comment by skyzh** [comment]
To other maintainers: this PR won’t be merged as we’ve decided to move to global topN. I’ll just make some comments and get it closed.

File: src/stream/src/executor/top_n_global.rs [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+            internal_key_indices,
+            internal_key_order_types,
+            total_count,
+        })
+    }
+    /// Sort `new_row` and keep only `self.limits` entries
+    fn sort_and_filter(&self, new_rows: &Vec<Row>, schema: &Schema) {
+        let limit = self.limit.unwrap_or(0);
+        let offset = self.offset;
+
```
Comment (skyzh) [inline src/stream/src/executor/top_n_global.rs:138]: You may refer to https://github.com/singularity-data/risingwave/blob/761ec04f9eb1296c326ffff0942cf4d9787aca6d/src/stream/src/executor/managed_state/top_n/top_n_state_new.rs#L115-L124 if you want to implement something similar in the future. We can get the storage pk from the deserialized row by re-using `get_topn_row`.

**Review by skyzh — COMMENTED** [review]
Generally looks good for the executor part. If you want to implement an executor in the future, this seems like a starting point, and could be your first PR. Then you should add unit tests, and add protobuf definition for the executor, so that it can be used and scheduled by meta service.

File: src/stream/src/executor/top_n_global.rs (hunk 1 above)
Comment (Shmiwy) [reply src/stream/src/executor/top_n_global.rs:138]: it seems that `row` have been sorted before?

Comment (Shmiwy) [reply src/stream/src/executor/top_n_global.rs:138]: and the `ordered` just flips  the order if needed？

Comment (skyzh) [reply src/stream/src/executor/top_n_global.rs:138]: You'll need to merge multiple batches of sorted `Row`s by users' specified order (like `a ASC, b DESC`). Therefore, to merge them, we'll need to use `OrderedRow` to flip the order of some columns, so as to produce correct result.

**Pull request closed without merging**